src/pebble.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""The agent pebble service module."""
+
+import logging
+import typing
+
+import ops
+
+import server
+from state import State
+
+logger = logging.getLogger(__name__)
+
+
+class PebbleService(ops.Object):
+    """The charm pebble service manager."""
+
+    def __init__(self, charm: ops.CharmBase, state: State):
+        """Initialize the pebble service.
+
+        Args:
+            charm: The parent Jenkins k8s agent charm.
+            state: The Jenkins k8s agent state.
+        """
+        super().__init__(charm, "pebble-service")
+        self.charm = charm
```

Comment:
`charm` is only used for the status and the get_container
If you descope these from this service you don't need the charm, which will greatly simplify the unittesting and avoid some sort of circular dependency

Reply:
That does make it much simpler, delegated all the status sets to the controller methods. Thank you!